renderdoc/driver/d3d12/d3d12_replay.cpp

```diff
@@ -1060,37 +1066,9 @@ void D3D12Replay::FillRegisterSpaces(const D3D12RenderState::RootSignature &root
           if(num == UINT_MAX)
           {
             // find out how many descriptors are left after
-            UINT availDescriptors = heap->GetNumDescriptors() - offset - UINT(e->offset);
-
-            // if this is an unbounded size table, find the matching bindpoint to try and
-            // upper-bound its size.
-            // due to D3D12's messed up resource matching, an array of descriptors in the root
-            // signature might match multiple bindings in the HLSL. So we just need to pick the
-            // highest-register item in the space and use its size.
-
-            const Bindpoint *bind = NULL;
-
-            // find the
-            for(const Bindpoint &b : mapping.readOnlyResources)
-            {
-              if(b.bindset == (int32_t)range.RegisterSpace && (bind == NULL || bind->bind < b.bind))
-                bind = &b;
-            }
-
-            // the size is however many registers are between the base of this range and that last
-            // bind, plus the size of the bind (at least 1 if it's not arrayed).
-            // If we didn't find any bind, clamp to 128 instead to prevent massive descriptor heaps
-            // from being passed through.
-            if(bind)
-            {
-              uint32_t arraySize = bind->arraySize == UINT_MAX ? 0U : bind->arraySize;
-              num = RDCMIN(availDescriptors,
-                           bind->bind - range.BaseShaderRegister + RDCMAX(1U, arraySize));
-            }
-            else
-            {
-              num = RDCMIN(availDescriptors, 128U);
-            }
+            // TODO: How do Bindpoints fit in when we fetch the root signature all at once?
+            // Do we check all shaders? Skip the Bindpoint check altogether?
```

Comment:
From what I recall the bindpoint check was still a very good win, in the case of e.g. a million descriptors in a heap but the shader declaring an array of `[4096]`. In this scenario I think we'd want to iterate over all shaders that the range is visible to and take the max size of them all.